### PR TITLE
Read resolution from env variable

### PR DESCRIPTION
**Pull request by pickworth**
@ashkulz here is the PR as discussed in issue #21 

Next step is I guess to change the https://github.com/wkhtmltopdf/wkhtmltopdf code to set these variables according to options passed in by --viewport-size argument, which would make for a cleaner implementation

Please let me know if any issues.

@blazerguns FYI

### CODE REVIEW

File: src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformScreenQt.cpp [hunk 1]
```diff
@@ -39,9 +39,38 @@
 #include "QWebPageClient.h"
 #include <QApplication>
 #include <QDesktopWidget>
+#include <stdlib.h>
 
 namespace WebCore {
 
+static unsigned long qt_get_screen_width() {
+  char *width = getenv("QT_SCREEN_WIDTH");
```
Comment (ashkulz) [inline src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformScreenQt.cpp]: should be `WKHTMLTOX_SCREEN_WIDTH` as this is not official Qt.

File: src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformScreenQt.cpp [hunk 2]
```diff
@@ -39,9 +39,38 @@
...
+  unsigned long screen_width = 1366;
+
+  if (width != NULL) {
+      unsigned long screen_width_tmp = atol(width);
```
Comment (ashkulz) [inline src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformScreenQt.cpp]: Use [`QString::toInt`](http://doc.qt.io/qt-4.8/qstring.html#toInt), as that does error checking as well.

**Comment by ashkulz** [comment]
Please address the above comments and then rebase/squash the changes on to the latest `wk_4.8.7` commit. Also, looks like `qt/src/3rdparty/webkit/Source/WebCore/platform/qt/ScreenQt.cpp` needs to be modified as well -- it got left out in previous PR.

File: src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformScreenQt.cpp (hunk 1 above)
Comment (pickworth) [inline src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformScreenQt.cpp]: Thanks, fixed

**Comment by pickworth** [comment]
@ashkulz Thanks for the feedback, will work on updating the PR some time soon

File: src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformScreenQt.cpp (hunk 2 above)
Comment (pickworth) [inline src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformScreenQt.cpp]: Changed to Use qgetenv() in stead of getenv()

**Comment by pickworth** [comment]
@ashkulz Please review changes and let me know if this is acceptable.

Everything is changed to qt library calls - using qgetenv() instead of getenv()

**Comment by blazerguns** [comment]
@ashkulz Once the changes are accepted, we will rebase and squash the commit. Please let us know if you are ok with it.

File: src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformScreenQt.cpp [hunk 3]
```diff
@@ -39,9 +39,40 @@
...
+  QByteArray width = qgetenv("WKHTMLTOX_SCREEN_WIDTH");
+  unsigned long screen_width = 1366;
+
+  bool ok;
+  unsigned long screen_width_tmp = width.toULong(&ok, 10);
+  if (ok == true) {
+      if ((screen_width_tmp >= 320) && (screen_width_tmp <= 7680)) {
+         return screen_width_tmp;
+      }
+  }
```
Comment (ashkulz) [inline src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformScreenQt.cpp]: Formatting is not correct (2 spaces vs 4 spaces). Also, would prefer simpler code similar to the following (untested):

``` c
static unsigned int wkhtmltox_screen_width()
{
    QByteArray override = qgetenv("WKHTMLTOX_SCREEN_WIDTH");

    bool ok;
    unsigned int width = override.toUInt(&ok);
    if (ok)
        return qMax(320, qMin(7680, width));

    return 1366; // default screen width
}
```

Screen resolution is not going to exceed `int` and ideally the function name should be similar to the environment variable.

**Comment by ashkulz** [comment]
Please fix the formatting and make changes as mentioned above. I'd prefer to review rebased commits, to avoid any merge conflicts later on. Otherwise, looks good to me :+1:

**Comment by ashkulz** [comment]
@blazerguns: there's a compilation failure, guess you forgot to verify the change:

```
wkhtmltopdf\qt\src\3rdparty\webkit\Source\WebCore\platform\qt\PlatformScreenQt.cpp(53) : error C2782: 'const T &qMin(const T &,const T &)' : template parameter 'T' is ambiguous
        wkhtmltopdf\static-build\msvc2013-win32-dbg\qt\include\qtcore\../../../../../qt/src/corelib/global/qglobal.h(1326) : see declaration of 'qMin'
        could be 'unsigned int'
        or       'int'
```

**Comment by blazerguns** [comment]
@ashkulz , Thank you for correcting the mistake on my behalf. Is there plan to roll out new npm release?

**Comment by blazerguns** [comment]
@ashkulz, Sorry please ignore the above comment. Its a mistake, my bad.

**Pull request closed without merging**